gdal/ogr/ogrsf_frmts/edigeo/ogredigeodatasource.cpp

```diff
@@ -1153,141 +1153,143 @@ int OGREDIGEODataSource::BuildLineStrings()
+                            break;
+                        }
+                        else if (*psNext == sSecondRing.back())
+                        {
+                            bFound = true;
+                            bReverseSecond = true;
+                            break;
+                        }
                     }
                 }
-            }
 
-            if( !bFound )
-            {
-                CPLDebug("EDIGEO", "Cannot find ring for FEA %s / PFE %s",
-                        osFEA.c_str(), osPFE.c_str());
-                break;
-            }
-            else
-            {
-                const xyPairListType& secondRing = *(aoPARPtrList[i]);
-                aoPARPtrList[i] = nullptr;
-                if( !bReverseSecond )
+                if( !bFound )
                 {
-                    for(i=1;i<(int)secondRing.size();i++)
-                        aoXY.push_back(secondRing[i]);
-                    psNext = &secondRing.back();
+                    CPLDebug("EDIGEO", "Cannot find ring for FEA %s / PFE %s",
+                                osFEA.c_str(), aosPFE[k].c_str());
+                    break;
                 }
                 else
                 {
-                    for(i=1;i<(int)secondRing.size();i++)
-                        aoXY.push_back(secondRing[secondRing.size()-1-i]);
-                    psNext = &secondRing[0];
+                    const xyPairListType& secondRing = *(aoPARPtrList[i]);
+                    aoPARPtrList[i] = nullptr;
+                    if( !bReverseSecond )
+                    {
+                        for(i=1;i<(int)secondRing.size();i++)
+                            aoXY.push_back(secondRing[i]);
+                        psNext = &secondRing.back();
+                    }
+                    else
+                    {
+                        for(i=1;i<(int)secondRing.size();i++)
+                            aoXY.push_back(secondRing[secondRing.size()-1-i]);
+                        psNext = &secondRing[0];
+                    }
                 }
+
+                nIter ++;
             }
 
-            nIter ++;
+            aoXYList.push_back(aoXY);
         }
 
-        aoXYList.push_back(aoXY);
-    }
-
-/* -------------------------------------------------------------------- */
-/*      Create feature.                                                 */
-/* -------------------------------------------------------------------- */
-    OGRFeature* poFeature = CreateFeature(osFEA);
-    if( poFeature )
-    {
-        std::vector<OGRGeometry*> aosPolygons;
-        for( int j = 0; j < (int)aoXYList.size(); j++ )
+    /* -------------------------------------------------------------------- */
+    /*      Create feature.                                                 */
+    /* -------------------------------------------------------------------- */
+        OGRFeature* poFeature = CreateFeature(osFEA);
```

Comment:
This will create one feature for each polygon of the multipolygon. Wouldn't it be better to create a single feature with a OGRMultiPolygon instead ?

Reply:
Yes indeed, that would be better! I'll modify that.

Reply:
However the other ways to record multipolygons also creates one feature for each polygon of the multipolygon. So it is not an major issue: right now it can be enough.

Reply:
> the other way to record multipolygons also creates one feature for each polygon of the multipolygon

I don't understand what you mean. If you put the std::vector<xyPairListType> aoXYList; declaration right before the outer loop, and the feature creation after it, then organizePolygons() will generate a multipolygon if needed

Reply:
I made the modification.